src/main/java/com/google/sps/servlets/CalendarServlet.java

```diff
@@ -123,4 +123,29 @@ public void doGet(
     }
     return events;
   }
+
+  /**
+   * Creates a new event when POST method called. Use the time boundaries given in the request for
+   * creating the event in the primary calendar. The summary for this event is Read emails.
+   *
+   * @param request Http request from the client. Body contains start and end parameters
+   * @param response 403 if user is not authenticated, or string saying the event has been created
+   * @throws IOException if an issue arises while processing the request
+   */
+  @Override
+  public void doPost(
+      HttpServletRequest request, HttpServletResponse response, Credential googleCredential)
+      throws IOException {
+    assert googleCredential != null
+        : "Null credentials (i.e. unauthenticated requests) should already be handled";
+
+    CalendarClient calendarClient = calendarClientFactory.getCalendarClient(googleCredential);
+    Date start = new Date(request.getParameter("start"));
+    Date end = new Date(request.getParameter("end"));
+    String summary = "Read emails";
```

Comment:
It's weird that the generic `CalendarServlet` always creates events with the summary "Read emails". I think that the client should supply this value. Of course you also need to make sure it's consistent with the event summary that `FreeTimeServlet` expects.

Reply:
I agree that this would give the user more control, but I don't know that we could capture all the events created in this process in `PlanMailServlet` (i.e. what happens if the user changes the summary, but we still need to account for events created with the previous summary?)

Reply:
I am not saying the user should specify the event summary. I am just saying it should come from the JS client. You can hardcode on the client (as a temporary solution) or have the PlanMailServlet tell the client what summary to use.

Reply:
I am now passing the event summary from `populatePlanMail`